src/ResourceParser.h

```diff
@@ -133,6 +133,13 @@ namespace snowcrash {
                             attributes.sourceMap.name.sourceMap = out.sourceMap.name.sourceMap;
                         }
                     }
+                    else {
+
+                        mdp::CharactersRangeSet sourceMap = mdp::BytesRangeSetToCharactersRangeSet(node->sourceMap, pd.sourceData);
+                        out.report.error = Error("attributes can not be added to a resource which doesn't have a name",
+                                                 BusinessError,
```

Comment:
`BusinessError` should not be used for these kind of issues. This is a regular `ApplicationError` or add a new one. `BusinessError` should be used only for "violations" against give "business" of operation e.g. trying to use unsupported tabs.